src/rules/function-calc-operator-space-before/__tests__/index.js

```diff
@@ -0,0 +1,46 @@
+import { ruleTester } from "../../../testUtils"
+import rule, { ruleName, messages } from ".."
+
+const testRule = ruleTester(rule, ruleName)
+
+testRule("always", tr => {
+  tr.ok("a {}")
+  tr.ok("a { color: color(red s(-10%)); }")
+  tr.ok("a { top: calc(1px + 2px); }")
+  tr.ok("a { top: calc(1px * 2); }")
+  tr.ok("a { top: calc(1px / 2); }")
+  tr.ok("a { top: calc(1px - 2px); }")
+  tr.ok("a { top: calc(1px +2px); }")
+  tr.ok("a { top: calc(calc(1px + 2px) + 3px); }")
+
+  tr.notOk("a { top: calc(1px+ 2px); }", messages.expectedBefore())
+  tr.notOk("a { top: calc(1px* 2); }", messages.expectedBefore())
+  tr.notOk("a { top: calc(1px/ 2); }", messages.expectedBefore())
+  tr.notOk("a { top: calc(1px- 2px); }", messages.expectedBefore())
+  tr.notOk("a { top: calc(1px+2px); }", messages.expectedBefore())
+  tr.notOk("a { top: calc(1px  + 2px); }", messages.expectedBefore())
+  tr.notOk("a { top: calc(1px\t+ 2px); }", messages.expectedBefore())
+  tr.notOk("a { top: calc(calc(1px+ 2px) + 3px); }", messages.expectedBefore())
+  tr.notOk("a { top: calc(calc(1px + 2px)+ 3px); }", messages.expectedBefore())
+})
+
+testRule("never", tr => {
```

Comment:
Not sure if this part is a good idea since something like `calc(1px+ 2px)` is invalid.